### PR TITLE
Add the jira card information to the test case

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Compared to previous version, update each test case to retrieve the associated jira card for tracking, using the parent test name

### Motivation
Improve the management of jira tickets for TPM

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/libs/junit_upload.py [hunk 1]
```diff
@@ -252,43 +258,40 @@ def repack_macos_junit_tar(infile, outfile):
...
-    Search in jira if a card already exist for the given testsuite
+    Search in jira if a card already exist for the given test
     """
-    from jira import JIRA
+    if test_name in jira_cache:
+        return jira_cache[test_name]
 
-    fake_jira_card = "XYZ-123"
-    tree = ET.parse(xmlfile_path)
+    jira_card = "XYZ-123"
```
Comment (KevinFairise2) [inline tasks/libs/junit_upload.py:268]: If I understand correctly, `XYZ-123` is returned only when we encounter an error somewhere. Could we return something like `ERROR: <info about error>` and parse the `ERROR` in the worklow, so we know we had an error and we also have some information about the error in the workflow? WDYT?

**Review by KevinFairise2 — APPROVED** [review]
Small suggestion, otherwise LGTM!

File: tasks/libs/junit_upload.py (hunk 1 above)
Comment (chouetz) [reply tasks/libs/junit_upload.py:268]: We can have 2 kind of errors:
- jira token not found (should not happen)
- any issue using the JIRA api
as a consequence I can maybe send `ERROR-TOKEN` or `ERROR-API` as jira card name to adapt my message in the workflow, wdyt?

Comment (KevinFairise2) [reply tasks/libs/junit_upload.py:268]: Yes, sounds good! This way it also more explicit in the code what this jira_card value refers to!

**Pull request merged**